code/modules/ruins/lavaland_ruin_code.dm

```diff
@@ -0,0 +1,213 @@
+/obj/structure/lavaland_door
+	name = "necropolis gate"
+	desc = "An imposing, seemingly impenetrable door."
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "door"
+	anchored = 1
+	density = 1
+	bound_width = 96
+	bound_height = 96
+	burn_state = LAVA_PROOF
+	luminosity = 1
+
+/obj/structure/lavaland_door/singularity_pull()
+	return 0
+
+/obj/structure/lavaland_door/Destroy()
+	return QDEL_HINT_LETMELIVE
+
+/obj/machinery/lavaland_controller
+	name = "weather control machine"
+	desc = "Controls the weather."
+	icon = 'icons/obj/machines/broadcast.dmi'
+	icon_state = "broadcaster"
+	var/ongoing_weather = FALSE
+	var/weather_cooldown = 0
+
+/obj/machinery/lavaland_controller/process()
+	if(ongoing_weather || weather_cooldown > world.time)
+		return
+	ongoing_weather = TRUE
+	weather_cooldown = world.time + rand(3500, 6500)
+	var/datum/weather/ash_storm/LAVA = new /datum/weather/ash_storm
+	LAVA.start()
+	ongoing_weather = FALSE
+
+/obj/machinery/lavaland_controller/Destroy()
+	return QDEL_HINT_LETMELIVE
+
+
+
+//lavaland_surface_seed_vault.dmm
+//Seed Vault
+
+/obj/effect/spawner/lootdrop/seed_vault
+	name = "seed vault seeds"
+	lootcount = 1
+
+	loot = list(/obj/item/seeds/gatfruit = 10,
```

Comment:
The gatfruit seed is not guaranteed, there is a chance it won't spawn in the vault ruin, then there is a chance the vault ruin won't spawn. The actual plant takes ages to grow too and needs to be watched constantly.